src/packet_analysis/protocol/ip/ConnKey.h

```diff
@@ -0,0 +1,107 @@
+// See the file "COPYING" in the main distribution directory for copyright.
+
+#pragma once
+
+#include <netinet/in.h>
+#include <optional>
+
+#include "zeek/Conn.h"
+#include "zeek/ConnKey.h"
+
+namespace zeek {
+
+namespace detail {
+
+/**
+ * Struct for embedding into an IPBasedConnKey.
+ */
+struct PackedConnTuple {
+    in6_addr ip1;
+    in6_addr ip2;
+    uint16_t port1 = 0;
+    uint16_t port2 = 0;
+    uint16_t proto = detail::INVALID_CONN_KEY_IP_PROTO;
+} __attribute__((packed, aligned));
+
+} // namespace detail
+
+/**
+ * Abstract key class for IP-based connections.
+ *
+ * ConnKey instances for IP always hold a ConnTuple instance which is provided
+ * by the IPBasedAnalyzer. The InitConnTuple() method stored a normalized version
+ * in the tuple, loosing the information about orig and responder.
+ */
+class IPBasedConnKey : public zeek::ConnKey {
+public:
+    /**
+     * Initializes the key to the given 5-tuple. This canonicalizes the
+     * packed tuple storage, including potential endpoint flips for
+     * consistent connection lookups regardless of directionality.
```

Comment:
I am not sire if this matters super much - but the methods in this file miss the documentation of arguments and return values that is required according to our style guide.

Reply:
Added them.